controller/src/main/java/io/pravega/controller/task/Stream/StreamMetadataTasks.java

```diff
@@ -706,6 +713,14 @@ public void setRequestEventWriter(EventStreamWriter<ControllerEvent> requestEven
                 stream, segmentCut.getKey(), segmentCut.getValue(), hostControllerStore, this.connectionFactory), executor));
     }
 
+    public CompletableFuture<Map<Integer, Long>> getSealedSegmentsSize(String scope, String stream, List<Integer> sealedSegments) {
+        return Futures.allOfWithResults(
+                sealedSegments
+                        .stream()
+                        .parallel()
+                        .collect(Collectors.toMap(x -> x, x -> getSegmentOffset(scope, stream, x))));
```

Comment:
Note that `getSegmentOffset` is computed like this:

```
return withRetries(() -> segmentHelper.getSegmentInfo(
                scope,
                stream,
                segmentNumber,
                hostControllerStore,
                this.connectionFactory), executor)
                .thenApply(WireCommands.StreamSegmentInfo::getSegmentLength);
```

which seems to be stale as the corresponding field seems to be `writeOffset` in `WireCommands`. Also, note that `writeOffset` is obtained via `SegmentProperties.getLength` and the javadoc of `getLength` says:

```
/**
     * Gets a value indicating the full, readable length of the StreamSegment. This includes the range of bytes that are
     * inaccessible due to them being before the StartOffset.
     */
```

Do we need to subtract the start offset to get the correct size value?

Reply:
>Do we need to subtract the start offset to get the correct size value?

No we dont because the logic is built around using overall size for a segment. 
It also works under the assumption that all truncation is performed via controller and it is never bypassed. 

So a controller has last truncation record and it has absolute sizes of all segments (sealed or active). 

Also, everytime we compute a stream cut we also record the "absolute" size corresponding to the stream cut. 
 
So computing for size based retention ==> given a stream cut, should we truncate at it or not:
```if (Current-Absolute-size MINUS size-at-stream-cut > configuration.retentionpolicy.size) {
// truncate at the given stream cut. 
}```

Reply:
Two points:

1- I think my original question above about `writeOffset` remains unanswered.
2- To reflect our offline discussion, the current policy implementation strives to maintain a minimum of `policy.size`.

Reply:
> 1- I think my original question above about writeOffset remains unanswered.
what is your question exactly? is it following?
> Do we need to subtract the start offset to get the correct size value?
then i have answered exactly that above. if its something else, could you please post the question. 

> To reflect our offline discussion, the current policy implementation strives to maintain a minimum of policy.size.
Yes, exactly

Reply:
Ok.